api/bus.go

```diff
@@ -485,6 +489,81 @@ type GougingSettings struct {
 	MinMaxEphemeralAccountBalance types.Currency `json:"minMaxEphemeralAccountBalance"`
 }
 
+// Types related to multipart uploads.
+type (
+	MultipartCreateRequest struct {
+		Bucket string `json:"bucket"`
+		Path   string `json:"path"`
+	}
+	MultipartCreateResponse struct {
+		UploadID string `json:"uploadID"`
+	}
+	MultipartAbortRequest struct {
+		Bucket   string `json:"bucket"`
+		Path     string `json:"path"`
+		UploadID string `json:"uploadID"`
+	}
+	MultipartAbortResponse struct {
+	}
+	MultipartCompleteRequest struct {
+		Bucket   string `json:"bucket"`
+		Path     string `json:"path"`
+		UploadID string `json:"uploadID"`
+		Parts    []MultipartCompletedPart
+	}
+	MultipartCompletedPart struct {
+		PartNumber int    `json:"partNumber"`
+		ETag       string `json:"etag"`
+	}
+	MultipartCompleteResponse struct {
+		ETag string `json:"etag"`
+	}
+	MultipartAddPartRequest struct {
+		Bucket        string                                   `json:"bucket"`
+		Etag          string                                   `json:"etag"`
+		Path          string                                   `json:"path"`
+		ContractSet   string                                   `json:"contractSet"`
+		UploadID      string                                   `json:"uploadID"`
+		PartialSlabs  []object.PartialSlab                     `json:"partialSlabs"`
+		PartNumber    int                                      `json:"partNumber"`
+		Slices        []object.SlabSlice                       `json:"slices"`
+		UsedContracts map[types.PublicKey]types.FileContractID `json:"usedContracts"`
+	}
+	MultipartListUploadsRequest struct {
+		Bucket         string `json:"bucket"`
+		Prefix         string `json:"prefix"`
+		KeyMarker      string `json:"keyMarker"`
+		UploadIDMarker string `json:"uploadIDMarker"`
+		Limit          int    `json:"limit"`
+	}
+	MultipartListUploadsResponse struct {
+		Uploads []MultipartListUploadItem `json:"uploads"`
+	}
+	MultipartListUploadItem struct {
```

Comment:
is `bucket` missing here?

Reply:
So it could be in `MultipartListUploadsResponse` but I didn't add it since we don't need it to satisfy the corresponding S3 endpoint. We can just take the bucket from the request.